syntax="proto2";
import "brpc/options.proto";

package brpc.policy;

message RdmaRpcMeta {
    optional RdmaRpcRequestMeta request = 1;
    optional RdmaRpcResponseMeta response = 2;
    optional int64 correlation_id = 3;
    optional int32 attachment_size = 4;
    optional bytes authentication_data = 5;
}

message RdmaRpcRequestMeta {
    required string service_name = 1;
    required string method_name = 2;
    optional int64 log_id = 3;
    optional int64 trace_id = 4;
    optional int64 span_id = 5;
    optional int64 parent_span_id = 6;
}

message RdmaRpcResponseMeta {
    optional int32 error_code = 1;
    optional string error_text = 2;
}
